src/sparseml/optim/manager.py

```diff
@@ -622,6 +622,45 @@ def phase(self, epoch: float) -> Optional[str]:
             else:
                 return "quantized_pruned"
 
+    def get_start_end_epochs(self) -> Dict[str, Tuple[float, float]]:
+        """
+        Return an OrderedDict mapping each stage to its min and max epoch. If not a
+        staged manager, map 'all' to the the min and max epochs
+        """
+        if isinstance(self.modifiers, List):
+            return OrderedDict({"all": (self.min_epochs, self.max_epochs)})
+        else:
+            stage_max_min = OrderedDict()
+            for stage, mod_list in self.modifiers.items():
+                epoch_floors = [
+                    math.floor(mod.start_epoch)
+                    for mod in mod_list
+                    if mod.start_epoch > -1
+                ] + [
+                    math.floor(mod.end_epoch) for mod in mod_list if mod.end_epoch > -1
+                ]
+                epoch_min = min(epoch_floors) if epoch_floors else -1
+
+                epoch_ceils = [
+                    math.ceil(mod.start_epoch)
+                    for mod in mod_list
+                    if mod.start_epoch > -1
+                ] + [math.ceil(mod.end_epoch) for mod in mod_list if mod.end_epoch > -1]
+                epoch_max = max(epoch_ceils) if epoch_ceils else -1
+
+                stage_max_min[stage] = (epoch_min, epoch_max)
```

Comment:
Since you are doing floor/ceil over both start/end epoch, this isn't technically the start & end epochs right? What's the reasoning for including both in epoch_min/epoch_max? Does something go wrong if you just include start_epoch for epoch_min and end_epoch for epoch_max?

Reply:
Good question and I'm not 100% sure on the answer. I copied the logic from the modifier props `min_epochs` and `max_epochs` and I assumed they're covering some edge cases, but a good edge case doesn't come to mind right now. 

Actually, a counter-example edge case for how this logic may break comes to mind. If we have a modifier with a positive value end epoch and no start epoch, we could return a value that doesn't make sense here

@dbogunowicz @bfineran what is the original logic meant to cover?